### DISCUSSION TITLE
proper?

### DISCUSSION THREAD

**Discussion by ghost** [h3rald/min#113] [Q&A] [answered]
A symbol `proper?` is needed for *manual constraint control*, i thikn. Here use cases of the symbol:

```
10 (natural) proper? ; -> True
-1 (natural) proper? ; -> False
```
I couldn't find a way to use `expect` for `proper?`.

**Comment by h3rald** [comment, marked as answer]
This shouldn't be necessary I think, because:

* Predicates are available for all primitive types
* `type?` is available for typed dictionaries
* typeclasses, by definition, must return a Boolean value, so you could simply write:

       -1 typeclass:natural
       ;outputs false

**Reply by ghost** [reply]
Oh really? So basically i must use `-3 type:natural`. Good, thanks!

**Reply by h3rald** [reply]
Yes sorry, `type:natural` for now, then from v0.31.0 it will be `typeclass:natural` 😬
